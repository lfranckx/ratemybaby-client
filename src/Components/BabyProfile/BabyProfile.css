.profile-container {
    background-color: white;
    color: #444;
    box-shadow: 1.5px 3px 5px grey;
    border-radius: 8px;
    width: 400px;
    padding-bottom: 5px;
    margin: 0 215px;
}

.baby-pic {
    width: 400px;
    height: 500px;
    object-fit: cover;
    margin: 0;
    border-top-left-radius: 8px;
    border-top-right-radius: 8px;
    box-shadow: 1.5px 3px 5px grey;
}

#babyprofilename {
    font-size: 22pt;
}

.name-age {
    text-align: left;
    padding: 10px;
    border-bottom: solid 0.5px lightgray;
}

.buttons {
    position: relative;
    right: 5px;
}

#cake {
    height: 30px;
    width: 30px;
    position: relative;
    top: 4px;
    right: 3px;
    margin-right: 5px;
}

#house {
    height: 25px;
    width: 25px;
    margin-right: 5px;
    position: relative;
    top: 2px;
}

.age {
    font-size: 26px;
    display: block;
}

.country {
    display: block;
}


.about {
    font-size: 14pt;
    text-align: left;
    padding: 15px 10px;
    border-bottom: solid 0.5px lightgray;
}

.rating {
    font-size: 25px;
    font-weight: bold;
    margin: 10px;
}

/* Button Styling */
.rate-button {
    background-color: white;
    border-radius: 50%;
    border: none;
    box-shadow: 1.5px 2px 3px grey;
    margin: 5px;
}

/* Image Styling */
.rate-icon {
    height: 50px;
    width: 50px;
    position: relative;
}

#dislike {
    top: 5px;
    left: .12px
}

#like {
    top: 2.5px;
    left: .2px;
}

#superlike {
    height: 50px;
    width: 50px;
    position: relative;
    left: 0.5px;
    right: 1.5px;
    bottom: 1.2px;
}

#superlikebutton {
    margin: 5px;
}

/* Rating Button Labels */
.rate-labels {
    position: relative;
    right: 5px;
    margin: 0 17px;
    color: grey;
}

#superlike-label {
    position: relative;
    right: 12px;
}

#like-label {
    position: relative;
    right: 12px;
}

/* Button Animations */
.rate-button:active {
    box-shadow: 1.5px 2px 3px grey;
    transform: translateY(2px);
}

.rate-button:hover {
    box-shadow: 2px 4px 3px grey;
}

.rate-icon:hover {
    animation: shake 0.76s cubic-bezier(.36,.07,.19,.97) both;
    transform: translate3d(0, 0, 0);
    backface-visibility: hidden;
    perspective: 1000px;
}

@keyframes shake {
    30%, 50%, 70% {
        transform: rotate(7deg);
    }

    40%, 60% {
        transform: rotate(-7deg);
    }
}

/* MEDIA QUERIES */
@media (max-width: 700px) {

    .profile-container {
        width: 320px;
        margin: 0px;
    }

    .baby-pic {
        width: 320px;
        height: 350px;
    }

    .profile-container h3 {
        font-size: 18pt
    }

    #cake {
        height: 20px;
        width: 20px;
    }
    
    #house {
        height: 15px;
        width: 15px;
        margin-right: 10px;
        top: 0;
    }

    .age {
        font-size: 16px;
    }

    .country {
        font-size: 16px;
    }

    .about {
        font-size: 16px;
    }

    .rating {
        font-size: 20px;
    }

    .rate-icon {
        height: 25px;
        width: 25px;
    }

    #dislike {
        bottom: 5px;
    }

    #superlike {
        height: 25px;
        width: 25px;
        top: 1px;
    }

    .rate-labels {
        font-size: 12px;
        margin: 0 11px;

    }
}

@media (max-width: 321px) {
    .profile-container {
        border-bottom-left-radius: 0;
        border-bottom-right-radius: 0;
    }
}