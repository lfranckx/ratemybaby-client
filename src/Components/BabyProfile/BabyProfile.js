import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import BabyContext from '../../Contexts/BabyContext'
import './BabyProfile.css'

export default class BabyProfile extends Component {

    static defaultProps = {
        match: { params: {} },
    }

    static contextType = BabyContext

    componentDidMount() {
        this.context.setBaby(this.props.baby)
    }

    handleLike = (baby) => {
        baby.total_score += 2
        baby.total_votes += 2
        this.context.updateBaby(baby)
    }

    handleSuperLike = (baby) => {
        baby.total_score += 5
        baby.total_votes += 5
        this.context.updateBaby(baby)
    }

    handleDislike = (baby) => {
        baby.total_votes += 2
        this.context.updateBaby(baby)
    }
    
    render() {
        const { baby } = this.props
        if (!baby) {
            return <div className='loading'>Loading...</div>
        }

        if (baby.image_url === "") {
            return baby.image_url === "https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/babydrawing.png"
        }
        
        let rating = (baby.total_score / baby.total_votes)
        let percent = rating * 100
        let roundPercent = Math.round(percent) + '%'

        return (
                <section className="profile-container">
                    <img src={baby.image_url} alt="baby" className="baby-pic" />
                    <div className="name-age">
                        <h3 id="babyprofilename">{baby.baby_name}</h3>
                        <span className="age">
                            <img src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/bdaycake.png" 
                                alt="birthday-cake" 
                                id="cake" />
                            {baby.age}
                        </span>
                        <span className="country">
                            <img src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/house.png" 
                                alt="house" 
                                id="house" />
                            {baby.country}
                        </span>
                    </div>
                    
                    <div className="about">{baby.about}</div>
                    <div className="rating">{roundPercent}</div>
                    <button 
                        className="rate-button buttons" 
                        onClick={() => {
                            this.handleDislike(baby)
                        }}
                    >
                        <Link to="/rate">
                                <img src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/dislike.png" 
                                    alt="dislike" 
                                    className="rate-icon" 
                                    id="dislike"/>
                        </Link>
                    </button>
                    <button 
                        className="buttons"
                        id="superlikebutton"
                        onClick={() => {
                            this.handleSuperLike(baby)
                        }}
                    >
                        <Link to="/rate">
                            <img src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/blueavocado.png" 
                                alt="superlike" 
                                className="rate-icon"
                                id="superlike"/>
                        </Link>
                    </button>
                    <button 
                        className="rate-button buttons"
                        onClick={() => {
                            this.handleLike(baby)
                        }}
                    >
                        <Link to="/rate">
                            <img src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/avo.png" 
                                alt="like" 
                                className="rate-icon" 
                                id="like"/>
                        </Link>
                    </button>
                    
                    <div className="labels-container">
                        <span className="rate-labels" id="dislike-label">Dislike</span>
                        <span className="rate-labels" id="superlike-label">Superlike</span>
                        <span className="rate-labels" id="like-label">Like</span>
                    </div>
                </section>
        )
    }
}