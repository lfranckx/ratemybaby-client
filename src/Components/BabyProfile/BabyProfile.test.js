import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BabyProfile from './BabyProfile';

describe(`BabyProfile component`, () => {
    const props = {
        baby: {
            "baby_name": "Test",
            "age": "7", 
            "age_format": "months",
            "country": "United States",
            "about": "Testing data",
            "image_url": "https://ratemybaby-images.s3-us-west-1.amazonaws.com/John.jpg",
            "total_score": "25",
            "total_votes": "35",
            "parent_id": "2"
        }
    };

    it(`renders a profile given props`, () => {
        const wrapper = shallow(<BabyProfile {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it(`renders loading without props`, () => {
        const wrapper = shallow(<BabyProfile />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});