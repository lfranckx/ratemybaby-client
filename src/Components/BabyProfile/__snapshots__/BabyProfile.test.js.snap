// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BabyProfile component renders a profile given props 1`] = `
<section
  className="profile-container"
>
  <img
    alt="baby"
    className="baby-pic"
    src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/John.jpg"
  />
  <div
    className="name-age"
  >
    <h2
      id="babyprofilename"
    >
      Test
    </h2>
    <span
      className="age"
    >
      <img
        alt="birthday-cake"
        id="cake"
        src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/bdaycake.png"
      />
      7
       
      months
    </span>
    <span
      className="country"
    >
      <img
        alt="house"
        id="house"
        src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/house.png"
      />
      United States
    </span>
  </div>
  <div
    className="about"
  >
    Testing data
  </div>
  <div
    className="rating"
  >
    71%
  </div>
  <button
    className="rate-button buttons"
    onClick={[Function]}
  >
    <Link
      to="/rate"
    >
      <img
        alt="dislike"
        className="rate-icon"
        id="dislike"
        src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/dislike.png"
      />
    </Link>
  </button>
  <button
    className="rate-button buttons"
    id="superlikebutton"
    onClick={[Function]}
  >
    <Link
      to="/rate"
    >
      <img
        alt="superlike"
        className="rate-icon"
        id="superlike"
        src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/blueavocado.png"
      />
    </Link>
  </button>
  <button
    className="rate-button buttons"
    onClick={[Function]}
  >
    <Link
      to="/rate"
    >
      <img
        alt="like"
        className="rate-icon"
        id="like"
        src="https://ratemybaby-images.s3-us-west-1.amazonaws.com/logos-icons/avo.png"
      />
    </Link>
  </button>
  <div
    className="labels-container"
  >
    <span
      className="rate-labels"
      id="dislike-label"
    >
      Dislike
    </span>
    <span
      className="rate-labels"
      id="superlike-label"
    >
      Superlike
    </span>
    <span
      className="rate-labels"
      id="like-label"
    >
      Like
    </span>
  </div>
</section>
`;

exports[`BabyProfile component renders loading without props 1`] = `
<div
  className="loading"
>
  Loading...
</div>
`;
